pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -416,12 +418,33 @@ func (ctx *context) getDIDDocAndConnection(pubDID string) (*did.Doc, *Connection
 
 	logger.Debugf("creating new '%s' did for connection", didMethod)
 
+	// get the route configs (pass empty service endpoint, as default servie endpoint added in VDRI)
+	serviceEndpoint, routingKeys, err := route.GetRouterConfig(ctx.routeSvc, "")
```

Comment:
Based on the conversation we had [here](https://github.com/hyperledger/aries-framework-go/pull/1103/files#r370323241), I strongly suggest we stop coupling protocol services together, starting with this. All didexchange requires is a `NewDID()` function - nothing more. It doesn't care about routing, it just cares about sending *a* DID document.

Reply:
Is this a blocker for this merge or can we handle it later (a TODO) ?

Reply:
an urgent TODO I guess

Reply:
Created an issue - https://github.com/hyperledger/aries-framework-go/issues/1107.